src/main/java/org/jenkinsci/plugins/workinghours/model/ExcludedDate.java

```diff
@@ -39,74 +40,39 @@
-
-    public String getJsonData() {
-        return jsonData;
-    }
+    private static final String FIELD_UTC_OFFSET = "utcOffset";
+    private static final String FIELD_TIMEZONE = "timezone";
+    private static final String FIELD_TYPE = "type";
+    private static final String FIELD_NAME = "name";
+    private static final String FIELD_START_DATE = "startDate";
+    private static final String FIELD_END_DATE = "endData";
+    private static final String FIELD_NO_END = "noENd";
+    private static final String FIELD_REPEAT = "repeat";
+    private static final String FIELD_REPEAT_COUNT = "repeatCount";
+    private static final String FIELD_REPEAT_PERIOD = "repeatPeriod";
+    private static final String FIELD_REPEAT_INTERVAL = "repeatInterval";
 
 
     /**
      * Constructs an ExcludedDate object using json.
      *
-     * @param jsonData Json data to deserialize to an ExcludedDate object.
+     * @param sourceJSON Json data to deserialize to an ExcludedDate object.
      */
     @DataBoundConstructor
-    public ExcludedDate(String jsonData) {
-        if (gson == null) {
-            gson = new GsonBuilder().create();
-        }
-        this.dataContainer = gson.fromJson(jsonData, DataContainer.class);
-        this.jsonData = jsonData;
-    }
-
-    public DataContainer getDataContainer() {
-        return dataContainer;
-    }
-
-
-    /**
-     * Descriptor class required for the UI.
-     */
-    @Extension
-    public static class DescriptorImpl extends Descriptor<ExcludedDate> {
-
-        /**
-         * Gets human readable name.
-         *
-         * @return human readable name.
-         */
-        @Override
-        public final String getDisplayName() {
-            return "ExcludedDate descriptor";
-        }
-
-        /**
-         * Validate the date.
-         *
-         * @param value value to validate.
-         * @return FormValidation indicating whether date is valid.
-         */
-        public final FormValidation doCheckDate(
-                @QueryParameter final String value) {
-            if (!value.isEmpty() && !DateTimeUtility.isValidDate(value)) {
-                return FormValidation.error("Invalid date");
-            }
-            return FormValidation.ok();
+    public ExcludedDate(JSONObject sourceJSON) {
+        this.utcOffset = sourceJSON.getInt(FIELD_UTC_OFFSET);
+        this.timezone = sourceJSON.getString(FIELD_TIMEZONE);
+        this.startDate = new Date(sourceJSON.getJSONObject(FIELD_START_DATE));
+        if (!sourceJSON.getJSONObject(FIELD_END_DATE).isEmpty()) {
+            this.endDate = new Date(sourceJSON.getJSONObject(FIELD_END_DATE));
         }
+        this.name = sourceJSON.getString(FIELD_NAME);
+        this.repeat = sourceJSON.getBoolean(FIELD_REPEAT);
+        this.repeatCount = sourceJSON.getInt(FIELD_REPEAT_COUNT);
+        this.repeatPeriod = sourceJSON.getInt(FIELD_REPEAT_PERIOD);
+        this.repeatInterval = sourceJSON.getInt(FIELD_REPEAT_INTERVAL);
     }
```

Comment:
we should probably add a private default constructor so it can only be created via this constructor

Reply:
Done.